symengine/ntheory.cpp

```diff
@@ -375,13 +372,10 @@ int factor(const Ptr<RCP<const Integer>> &f, const Integer &n, double B1)
     if (mp_perfect_power_p(_n)) {
 
         unsigned long int i = 1;
-        integer_class m, rem;
+        integer_class rem;
         rem = 1; // Any non zero number
-        m = 2;   // set `m` to 2**i, i = 1 at the begining
 
-        // calculate log2n, this can be improved
-        for (; m < _n; ++i)
-            m = m * 2;
+        i = (std::log(mp_get_d(_n)) / std::log(2));
```

Comment:
it might overflow and give a different result...So should we use mp_get_ui here then ?